src/util/strencodings.cpp

```diff
@@ -128,6 +127,25 @@ void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut)
     }
 }
 
+void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut)
+{
+    std::string portStr;
+    SplitHostPort(in, portStr, hostOut);
+
+    uint16_t n;
+    if (!portStr.empty() && ParseUInt16(portStr, &n))
```

Comment:
The value `portOut` will be undefined behavior if `portStr = "-1"` as `ParseUInt16` will return false before assigning a value to `n`

Reply:
I don't know if I agree with you on this point:
My intention was not to change the behaviour of `SplitHostPort(...)` but rather to gain access to the port portion of a socket address as a string.
If you for example take a look at [src/httpserver.cpp](https://github.com/bitcoin/bitcoin/blob/a748782a11f0b23f8dcdf167f45d68c863524887/src/httpserver.cpp#L307&L308&L309) in `HTTPBindAddresses(...)` this function is used as follows:
`uint16_t port{some_default_port};
SplitHostPort(..., port, ...);`
The same can be seen in `src/bitcoin-cli.cpp` and `src/net.cpp`: the callers already set default port value before calling this function.

Reply:
That makes sense thanks!